packages/lib/pcd-collection/src/PCDCollection.ts

```diff
@@ -336,22 +360,52 @@ export class PCDCollection {
     } satisfies SerializedPCDCollection);
   }
 
-  public async deserialize(serialized: SerializedPCD): Promise<PCD> {
+  public async deserialize(
+    serialized: SerializedPCD,
```

Comment:
a thought that occurred to me is that to get to this point, the serialized PCD collection in localstorage must `JSON.parse` properly, which is perhaps an edge case not worth considering, just thought I'd bring it up. Another potential problem that this deserialization function is not robust to is if the serialized PCD collection contains malformed `SerializedPCD`s, e.g. if they were tampered with and someone inserted an object into that serialized PCD array that does not conform to the `SerializedPCD` interface.

I think these problems are probably beyond the scope if this PR, as my understanding is that the `UnknownPCD` aims to catch serialization incompatibilities / missing pcd packages more-so than anything else.

Reply:
Yeah, I agree this PR doesn't handle those cases, but that they're out of scope.  The UnknownPCD design is about isolating failures and incompatibilities in individual PCD packages.  The PCDCollection serialization and SerializedPCD format are part of the framework, so harder to isolate.  The goal here is to let "important functionality" (i.e. tickets) keep working when there's a breakage in less-important functionality.

If the whole PCDCollection can't be parsed, then I'd argue that an empty Zupass doesn't have any important functionality to maintain.  I could imagine a fallback where all the PCDs go missing, but new ones can be fetched from feeds, but that's a very different design challenge to implement in a way which is recoverable (if you want the malformed data to ever come back).

The SerializedPCD format is simple enough that I'm largely willing to assume that it's not going to get screwed up.  We can revisit that if/when it ever changes.